StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1211UsingAliasDirectivesMustBeOrderedAlphabeticallyByAliasName.cs

```diff
@@ -47,54 +49,66 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleCompilationUnit, SyntaxKind.CompilationUnit);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleNamespaceDeclaration, SyntaxKind.NamespaceDeclaration);
         }
 
-        private void HandleUsingDirectiveSyntax(SyntaxNodeAnalysisContext context)
+        private static void HandleCompilationUnit(SyntaxNodeAnalysisContext context)
         {
-            UsingDirectiveSyntax syntax = context.Node as UsingDirectiveSyntax;
-            if (syntax.Alias?.Name?.IsMissing != false)
+            var compilationUnit = context.Node as CompilationUnitSyntax;
+            if (compilationUnit != null)
             {
-                return;
+                HandleUsingDirectives(context, compilationUnit.Usings);
             }
+        }
 
-            CompilationUnitSyntax compilationUnit = syntax.Parent as CompilationUnitSyntax;
-            SyntaxList<UsingDirectiveSyntax>? usingDirectives = compilationUnit?.Usings;
-            if (!usingDirectives.HasValue)
+        private static void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDeclaration = context.Node as NamespaceDeclarationSyntax;
+            if (namespaceDeclaration != null)
             {
-                NamespaceDeclarationSyntax namespaceDeclaration = syntax.Parent as NamespaceDeclarationSyntax;
-                usingDirectives = namespaceDeclaration?.Usings;
+                HandleUsingDirectives(context, namespaceDeclaration.Usings);
             }
+        }
 
-            if (!usingDirectives.HasValue)
-            {
-                return;
-            }
+        private static void HandleUsingDirectives(SyntaxNodeAnalysisContext context, SyntaxList<UsingDirectiveSyntax> usingDirectives)
+        {
+            var usingAliasNames = new List<string>();
+            UsingDirectiveSyntax prevAliasUsingDirective = null;
 
             foreach (var usingDirective in usingDirectives)
             {
-                // we are only interested in nodes before the current node
-                if (usingDirective == syntax)
-                {
-                    continue;
-                }
-
                 // only interested in using alias directives
                 if (usingDirective.Alias?.Name?.IsMissing != false)
                 {
                     continue;
                 }
 
-                string alias = syntax.Alias.Name.ToString();
-                string precedingAlias = usingDirective.Alias.Name.ToString();
-                if (string.Compare(alias, precedingAlias, StringComparison.OrdinalIgnoreCase) >= 0)
+                string currentAliasName = usingDirective.Alias.Name.ToString();
+                if (prevAliasUsingDirective != null)
                 {
-                    continue;
+                    string prevAliasName = prevAliasUsingDirective.Alias.Name.ToString();
```

Comment:
:exclamation: I think this needs to use `ValueText` as well. See the previous remark.